### PR TITLE
Add a config plugin and docs for secure-store requiresAuthentication

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

Related to: https://github.com/expo/expo/issues/20030
Doesn't exactly fix the crash, mostly adds an easy way to fix it in release + docs.

@alanjhughes I don't see an easy way to check if the permission is there that will work for existing keys, or even for new ones. What do you think?

# How

Adapted from `expo-camera`.

# Test Plan

Tested manually by copying the module to a clean folder and ensuring the key is being set correctly.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by alanjhughes** [comment]
@aleqsio - We could add this to the `set` function? It would at least prevent the crash in expo go. You'll never be able to set a value that requires authentication in go so trying to retrieve one should not happen. I've tested this quickly in go and it stops the crash but maybe I'm not thinking of something?

```swift
if !options.requireAuthentication {
    query[kSecAttrAccessible as String] = accessibility
} else {
  guard let _ = Bundle.main.infoDictionary?["NSFaceIDUsageDescription"] as? String else {
    throw MissingPlistKeyException()
  }
  let accessOptions = SecAccessControlCreateWithFlags(kCFAllocatorDefault, accessibility, 
    SecAccessControlCreateFlags.biometryCurrentSet, nil)
  query[kSecAttrAccessControl as String] = accessOptions
}
```
The exception could be something like this
```swift
internal class MissingPlistKeyException: Exception {
  override var reason: String {
    "You must set `NSFaceIDUsageDescription` in your info.plist. This is not available in Expo go"
  }
}
```

**Comment by aleqsio** [comment]
that doesn't prevent crashes for keys that already were set before turning on face id, but I guess it's good enough 😅

File: packages/expo-secure-store/src/SecureStore.ts [hunk 1]
```diff
@@ -74,6 +74,10 @@ export type SecureStoreOptions = {
    * - iOS: Equivalent to [`kSecAccessControlBiometryCurrentSet`](https://developer.apple.com/documentation/security/secaccesscontrolcreateflags/ksecaccesscontrolbiometrycurrentset/).
    * Complete functionality is unlocked only with a freshly generated key - this would not work in tandem with the `keychainService`
    * value used for the others non-authenticated operations.
+   *
+   * Warning: This option is not supported in Expo Go when biometric authentication is available due to a missing NSFaceIDUsageDescription.
+   * In release builds or when using continous native generation, make sure to use the `expo-secure-store` config plugin.
```
Comment (alanjhughes) [inline packages/expo-secure-store/src/SecureStore.ts:79]:
```suggestion
   * In release builds or when using continuous native generation, make sure to use the `expo-secure-store` config plugin.
```

File: docs/pages/versions/unversioned/sdk/securestore.mdx [hunk 2]
```diff
@@ -14,6 +14,8 @@ import { SnackInline } from '~/ui/components/Snippet';
 
 **Size limit for a value is 2048 bytes. An attempt to store larger values may fail. Currently, we print a warning when the limit is reached, however, in a future SDK version an error might be thrown.**
 
+**The `requireAuthentication` option is not supported in Expo Go when biometric authentication is available due to a missing NSFaceIDUsageDescription.**
```
Comment (alanjhughes) [inline docs/pages/versions/unversioned/sdk/securestore.mdx:17]:
```suggestion
**The `requireAuthentication` option is not supported in Expo Go when biometric authentication is available due to a missing `NSFaceIDUsageDescription` key.**
```

**Review by alanjhughes — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]
Looks good, just update the build files once again because the SDK workflow is failing 😅

**Pull request merged**